packages/grafana-ui/src/components/TimePicker/TimePickerContent/TimePickerFooter.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC, useState, useCallback } from 'react';
+import { css, cx } from 'emotion';
+import { TimeZone, GrafanaTheme, getTimeZoneInfo } from '@grafana/data';
+import { stylesFactory, useTheme } from '../../../themes';
+import { TimeZoneTitle } from '../../TimeZonePicker/TimeZoneTitle';
+import { TimeZoneDescription } from '../../TimeZonePicker/TimeZoneDescription';
+import { TimeZoneOffset } from '../../TimeZonePicker/TimeZoneOffset';
+import { Button } from '../../Button';
+import { TimeZonePicker } from '../../TimeZonePicker/TimeZonePicker';
+
+interface Props {
+  timeZone?: TimeZone;
+  timestamp: number;
+  onChangeTimeZone: (timeZone: TimeZone) => void;
+}
+
+export const TimePickerFooter: FC<Props> = props => {
+  const { timeZone, timestamp, onChangeTimeZone } = props;
+  const [isEditing, setEditing] = useState(false);
+
+  const onToggleChangeTz = useCallback(
+    (event?: React.MouseEvent) => {
+      if (event) {
+        event.stopPropagation();
+      }
+      setEditing(!isEditing);
+    },
+    [isEditing, setEditing]
+  );
+
+  const theme = useTheme();
+  const style = getStyle(theme);
+
+  if (!timeZone) {
+    return null;
+  }
+
+  const info = getTimeZoneInfo(timeZone, timestamp);
+
+  if (!info) {
+    return null;
+  }
+
+  if (isEditing) {
```

Comment:
I wonder if you could use `ValuePicker` component from grafana-ui, as it actually implements this logic.

Reply:
Nice, will have a look :)

Reply:
It is a interesting idea. It would be nice but at the same time they are behaving a little bit different.